File: sorting-algos/selection_sort/selection_sort.c
// selection sort
#include<stdio.h>
void main()
{
    int n,i, j, min,temp=0;
    printf("Enter Size of an Array : ");
    scanf("%d",&n);
    int a[n];
    for ( i=0; i< n; i++)                                 // Input in Array
    {
        printf("Enter Element Number %d :",(i+1));
        scanf("%d",&a[i]);
    }
    for ( i=0; i<n-1; i++ )
    {
        min = i;
        for ( j=i+1; j<=n; j++)
        {
            if ( a[j] < a[min] )             // Swapping two elements.
            {
                min = j;
            }
        }
        temp = a[i];
        a[i] = a[min];
        a[min] = temp;
    }
    printf("Sorted Array --> \n");
    for ( i=0; i<n; i++ )
        printf("%d ",a[i]);
}

File: sorting-algos/selection_sort/selection_sort.py
#********** SELECTION_SORT *********
# enter n value and enter that many number in your list
# in-place sorting algorithm
# The smallest element is selected from the unsorted list 
# and swapped with the leftmost element, and that element becomes a part of the sorted list.
# This continues until whole list is sorted 
# Input: Value of n, List Elements
# Output: Unsorted list, Sorted List

l = []
n = int(input("Enter no. of elements in list (n): "))

for i in range(n):
    l.append(int(input()))

print("Unsorted list: ")
print(l)

MIN = l[0]

for i in range(n-1):
    
    MIN = i
    
    for j in range(i+1,n):
        if l[j] < l[MIN]:
            temp = l[j]
            l[j] = l[MIN]
            l[MIN] = temp

print("Sorted List: ")
print(l)